lib/gc/Transforms/TilingUtil.hpp

```diff
@@ -0,0 +1,26 @@
+//===- Tilig.hpp - Tiling ops using TilingInterface -------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEMPORARY_TILEUSINGINTERFACE_X_H
+#define TEMPORARY_TILEUSINGINTERFACE_X_H
+
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Interfaces/TilingInterface.h"
+#include <optional>
+namespace mlir {
+namespace linalgX {
+
+FailureOr<linalg::ForallReductionTilingResult> tileReductionUsingForall(
```

Comment:
What's the difference between this vs upstream one?

Reply:
The upstream one could only support do the K slicing for the linalg op with only one reduction iterator and not support do it for blocking layout with two reduction iterators(MKmk). Here is an enhancement for that.

Reply:
Thanks, better to add a notes for the difference here.

Reply:
Added